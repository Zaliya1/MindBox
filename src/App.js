import { useState } from "react"
import ToDoForm from "./components/ToDoForm"
import UncompletedTasks from "./components/UncompletedTasks"
import CompletedTasks from "./components/CompletedTasks"
import "./styles/App.css"

function App() {
  const [todos, setTodos] = useState([])
  const addTask = (userInput) => {
    if (userInput) {
      const newItem = {
        id: Date.now(),
        task: userInput,
        complete: false,
      }
      setTodos([...todos, newItem])
    }
  }
  const removeTask = (id) => {
    setTodos([...todos.filter((todo) => todo.id !== id)])
  }
  const toggleTask = (id) => {
    setTodos([
      ...todos.map((todo) =>
        todo.id === id ? { ...todo, complete: !todo.complete } : { ...todo }
      ),
    ])
  }
  return (
    <div className="App">
      <header className="header">
        <h1 className="heading">Список дел</h1>
      </header>
      <ToDoForm addTask={addTask} key="form" />
      <div className="tasks">
        <div className="uncompleted-tasks">
          <h3 className="heading-task">Задачи в процессе выполнения</h3>
          {todos.map((todo) => {
            if (!todo.complete)
              return (
                <ul>
                  <UncompletedTasks
                    todo={todo}
                    key={todo.id}
                    removeTask={removeTask}
                    toggleTask={toggleTask}
                  />
                </ul>
              )
            return null
          })}
        </div>
        <div className="completed-tasks">
          <h3 className="heading-task">Выполненные задачи</h3>
          {todos.map((todo) => {
            if (todo.complete)
              return (
                <ul>
                  <CompletedTasks
                    todo={todo}
                    key={todo.id}
                    removeTask={removeTask}
                    toggleTask={toggleTask}
                  />
                </ul>
              )
            return null
          })}
        </div>
      </div>
    </div>
  )
}

export default App
