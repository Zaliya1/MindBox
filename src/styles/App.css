* {
   margin: 0 auto;
   padding: 0;
   box-sizing: border-box;
}
button {
    cursor: pointer;
}
ul {
    list-style-type: none;
}
.App {
    max-width: 1000px;
}
.header {
    margin-bottom: 20px;
}
.heading {
    background-image: url(../images/img2.png);
    width: 500px;
    height: 340px;
    background-repeat: no-repeat;
    color: #3F3D56;
    font-weight: 400;
    padding-top: 68px;
    padding-left: 18px;
}
.input {
    border-color: #E5E5E5;
    color: #3F3D56;
    padding: 11px 20px;
    min-width: 500px;
    border-radius: 5px;
    font-size: 17px;
}
.btn-submit {
    background-color: #5D5FEF;
    color: white;
    border: 0;
    font-weight: 700;
    padding: 11px 20px;
    border-radius: 15px;
    font-size: 17px;
    transition: 1s ease background;
}
.btn-submit:hover {
    background-color: #181a70;
}
.form {
    display: flex;
    justify-content: space-evenly;
    margin-bottom: 20px;
}
.tasks {
    display: flex;
    justify-content: space-between;
    margin-bottom: 20px;
}
.heading-task {
    color: #3F3D56;
    text-align: center;
    margin-bottom: 10px;
}
.uncompleted-tasks, .completed-tasks {
    margin: 0;
    width: 47%;
}
.item-delete, .item-toggle {
    background-color: white;
    border: 0;
    align-items: center;
    width: 24px;
}
.task {
    border: 1px solid grey;
    padding: 10px;
    border-radius: 16px;
    margin-bottom: 5px;
    display: flex;
    justify-content: space-between;
    align-items: center;
}
.task-name {
    width: 90%;
}
.item-toggle:hover svg path {
    fill: #181a70;
}