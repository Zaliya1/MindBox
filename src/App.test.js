import { render, screen } from '@testing-library/react'
import App from './App'

test('renders learn react link', () => {
    render(<App />)
    const myList = screen.getByText(/список дел/i)
    const btn = screen.getByRole('button')
    const completedTasks = screen.getByText(/задачи в процессе выполнения/i)
    const uncompletedTasks = screen.getByText(/выполненные задачи/i)
    expect(myList).toBeInTheDocument()
    expect(btn).toBeInTheDocument()
    expect(completedTasks).toMatchSnapshot()
    expect(uncompletedTasks).toMatchSnapshot()
})