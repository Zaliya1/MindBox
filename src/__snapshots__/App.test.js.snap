// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<h3
  class="heading-task"
>
  Задачи в процессе выполнения
</h3>
`;

exports[`renders learn react link 2`] = `
<h3
  class="heading-task"
>
  Выполненные задачи
</h3>
`;
